htdocs/includes/odtphp/Segment.php

```diff
@@ -117,14 +117,14 @@ public function merge()
                 // Remove the IF tag
                 $this->xml = str_replace('[!-- IF '.$key.' --]', '', $this->xml);
                 // Remove everything between the ELSE tag (if it exists) and the ENDIF tag
-                $reg = '@(\[!--\sELSE\s' . $key . '\s--\](.*))?\[!--\sENDIF\s' . $key . '\s--\]@smU'; // U modifier = all quantifiers are non-greedy
+                $reg = '@(\[!--\sELSE\s' . preg_quote($key) . '\s--\](.*))?\[!--\sENDIF\s' . preg_quote($key) . '\s--\]@smU'; // U modifier = all quantifiers are non-greedy
```

Comment:
If you use preg_quote, you must use the second parameter to be sure result is the one expected.
Can you replace with
`preg_quote($key, '@') `
and tell if solution still works...

Reply:
I tried, it works. And it would also work if the constant name contains a `@`. Thanks for the reminder.